lektor/admin/static/js/widgets/ToastEditor.jsx

```diff
@@ -0,0 +1,213 @@
+  'code',
+  'codeblock'
+]
+
+class ToastEditor extends Component {
+  constructor () {
+    super()
+    this.element = null
+    this.count = 0
+    this.content = null
+
+    this.onChange = this.onChange.bind(this)
+    this.onLoad = this.onLoad.bind(this)
+    this.onRef = this.onRef.bind(this)
+    this.blur = this.blur.bind(this)
+    this.focus = this.focus.bind(this)
+
+    this.state = {
+      style: {},
+      editor: null
+    }
+  }
+
+  // from http://youmightnotneedjquery.com/#outer_height_with_margin
+  outerHeight (el) {
+    var height = el.offsetHeight
+    var style = window.getComputedStyle(el)
+
+    height += parseInt(style.marginTop) + parseInt(style.marginBottom)
+    return height
+  }
+
+  onLoad (editor) {
+    // add in the mode switcher
+    let toolbar = editor.getUI().getToolbar()
+    toolbar.insertItem(0, { type: 'button', options: { name: 'richtext-tab', text: 'Rich Text', event: 'mode-tab-richtext', tooltip: 'Use Rich Text Editor', className: 'mode-tab' } })
+    toolbar.insertItem(1, { type: 'button', options: { name: 'markdown-tab', text: 'Markdown', event: 'mode-tab-markdown', tooltip: 'Use Markdown Editor', className: 'mode-tab' } })
+    toolbar.insertItem(2, { type: 'divider', options: { name: 'switch-divider', className: 'mode-tab-divider' } })
+    let richtextbtn = toolbar.getItem(0).$el
+    let markdownbtn = toolbar.getItem(1).$el
+    editor.eventManager.addEventType('mode-tab-richtext')
+    editor.eventManager.addEventType('mode-tab-markdown')
+    editor.eventManager.listen('mode-tab-richtext', () => { markdownbtn.removeClass('active'); richtextbtn.addClass('active'); editor.changeMode('wysiwyg', true) })
+    editor.eventManager.listen('mode-tab-markdown', () => { richtextbtn.removeClass('active'); markdownbtn.addClass('active'); editor.changeMode('markdown', true) })
+    let activebtn = (this.props.type.default_view === 'richtext') ? richtextbtn : markdownbtn
+    activebtn.addClass('active')
+
+    this.recalculateHeight(editor)
+  }
+
+  onImageUpload (file, cb, source) {
+    // do image upload of file here, call cb once done
+    cb(file.name) // leave second param null to use user-specified text
+  }
+
+  recalculateHeight (editor) {
+    let currentEditor = editor.getCurrentModeEditor()
+    let editorHeight
+    try {
+      // markdown
+      let editorEl = currentEditor.editorContainerEl.getElementsByClassName('CodeMirror-sizer')[0]
+      editorHeight = this.outerHeight(editorEl)
+    } catch (e) {
+      // wysiwyg
+
+      // get height of all children in editor
+      let editorEl = currentEditor.$editorContainerEl[0].firstChild
+      let editorChildren = Array.from(editorEl.children)
+      let editorChildrenHeights = editorChildren.map(el => this.outerHeight(el))
+      editorHeight = editorChildrenHeights.reduce((a, b) => a + b)
+
+      // add on padding/border/margin of editor element
+      let marginTop = parseInt(window.getComputedStyle(editorEl).marginTop)
```

Comment:
Use `const` in as many places as possible. If a value never changes `const` will tell the reader that it remains unchanged. This goes for the entire PR.

Reply:
Fixed in 9c2ac2c